### PR TITLE
TOOLS-72 TOOLS-75 external tools validation

### PR DESCRIPTION
**Pull request by psihachina**
This commits adds:
- external tools validation 
- installing SSM Agent automatically 

Example: 

Windows 11 SSM Agent
image

Ubuntu (wsl2) SSM Agent
[asciicast](https://asciinema.org/a/6LgGPojgEgBlNsAykFwnoUvBt)

Ubuntu (wsl2) Docker
[asciicast](https://asciinema.org/a/8IozlqwC52HZ3Nt8qWBhlwrFO)

Windows 11 Docker
image

### Not tested on MasOS

### CODE REVIEW

File: internal/commands/base.go [hunk 1]
```diff
@@ -172,6 +175,84 @@ func (cc *izeBuilderCommon) Init() error {
 	viper.SetDefault("TF_LOG", fmt.Sprintf(""))
 	viper.SetDefault("TF_LOG_PATH", fmt.Sprintf("%v/tflog.txt", viper.Get("ENV_DIR")))
 
+	//Check Docker and SSM Agent
+	_, err = CheckCommand("docker", []string{"info"})
+	if err != nil {
+		return errors.New("docker is not running or is not installed")
```
[the remark this answers is not in the document]
Comment (AutomationD) [reply internal/commands/base.go:181]: Also let's provide to ssm plug-in installation a link in the place where we notify user to install the plugin manually.

**Review by AutomationD — CHANGES_REQUESTED** [review]
Looks good, minor changes requested.

File: internal/commands/base.go [hunk 2]
```diff
@@ -172,6 +175,84 @@ func (cc *izeBuilderCommon) Init() error {
...
+		err = InstallSSMAgent()
+		if err != nil {
+			return fmt.Errorf("install SSM Agent error: %v", err)
+		}
+
+		pterm.Success.Println("Installing SSM Agent")
+
+		err = CleanupSSMAgent()
+		if err != nil {
+			return fmt.Errorf("cleanup SSM Agent error: %v", err)
```
[the remark this answers is not in the document]
Comment (psihachina) [reply internal/commands/base.go:245]: @AutomationD https://staticcheck.io/docs/checks#ST1005

File: internal/commands/base_init.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ func (c *izeBuilderCommon) initConfig(filename string) (*config.Config, error) {
 func (c *izeBuilderCommon) initConfigPath(filename string) (string, error) {
 	path, err := config.FindPath(filename)
 	if err != nil {
-		return "", fmt.Errorf("Error looking for a Ize configuration: %s", err)
+		return "", fmt.Errorf("error looking for a Ize configuration: %s", err)
```
[the remark this answers is not in the document]
Comment (psihachina) [reply internal/commands/base_init.go:26]: @AutomationD https://staticcheck.io/docs/checks#ST1005

**Review by AutomationD — APPROVED** [review]
LGTM

**Pull request merged**